lib/AST.cpp

```diff
@@ -25,6 +27,7 @@ namespace pillar
 {
   namespace ast
   {
+    std::vector<std::function<void(void)>> *AST::lift_queue = new std::vector<std::function<void(void)>>();
```

Comment:
make this a member/field of the `AST` class. You don't need to make it heap-allocated either, just `std::vector<std::function<void(void)>> lift_queue;` will do.